lib/ts/recipe/webauthn/recipe.ts

```diff
@@ -0,0 +1,365 @@
+                    } else {
+                        // This factor hasn't been set up, we list all emails belonging to the user
+                        if (
+                            orderedLoginMethodsByTimeJoinedOldestFirst.some(
+                                (lm) => lm.email !== undefined && !isFakeEmail(lm.email)
+                            )
+                        ) {
+                            // If there is at least one real email address linked to the user, we only suggest real addresses
+                            result = orderedLoginMethodsByTimeJoinedOldestFirst
+                                .filter((lm) => lm.email !== undefined && !isFakeEmail(lm.email))
+                                .map((lm) => lm.email!);
+                        } else {
+                            // Else we use the fake ones
+                            result = orderedLoginMethodsByTimeJoinedOldestFirst
+                                .filter((lm) => lm.email !== undefined && isFakeEmail(lm.email))
+                                .map((lm) => lm.email!);
+                        }
+                        // We handle moving the session email to the top of the list later
+
+                        // Since in this case emails are not guaranteed to be unique, we de-duplicate the results, keeping the oldest one in the list.
+                        // The Set constructor keeps the original insertion order (OrderedByTimeJoinedOldestFirst), but de-duplicates the items,
+                        // keeping the first one added (so keeping the older one if there are two entries with the same email)
+                        // e.g.: [4,2,3,2,1] -> [4,2,3,1]
+                        result = Array.from(new Set(result));
+                    }
+
+                    // If the loginmethod associated with the session has an email address, we move it to the top of the list (if it's already in the list)
+                    if (sessionLoginMethod.email !== undefined && result.includes(sessionLoginMethod.email)) {
+                        result = [
+                            sessionLoginMethod.email,
+                            ...result.filter((email) => email !== sessionLoginMethod!.email),
+                        ];
+                    }
+
+                    // If the list is empty we generate an email address to make the flow where the user is never asked for
+                    // an email address easier to implement. In many cases when the user adds an email-password factor, they
+                    // actually only want to add a password and do not care about the associated email address.
+                    // Custom implementations can choose to ignore this, and ask the user for the email anyway.
+                    if (result.length === 0) {
+                        result.push(`${sessionRecipeUserId.getAsString()}@stfakeemail.supertokens.com`);
+                    }
```

Comment:
maybe we could chat about this.

Reply:
the flow is correct, but:
when using webauthn as an second factor and the first factor doesn't have an email adress, the credential generation would use the webuathn fake email as display - which would look unprofessional. because of this we could use display name as a paramenter for register options and default it to a sane value if its missing.

Reply:
done.